packages/utils/src/types.ts

```diff
@@ -14,3 +14,9 @@ export type RecursivePartial<T> = {
 export function bnToNum(bn: bigint): number {
   return Number(bn);
 }
+
+export type NonEmptyArray<T> = [T, ...T[]];
+
+export type ArrayToTuple<Tuple extends NonEmptyArray<unknown>> = {
```

Comment:
Can you add jsdoc comments to these types here?

Eg: ArrayToTuple converts an Array, eg: [1, 2, 3] from type number[] to [number, number, number]

Reply:
```suggestion
export type NonEmptyArray<T> = [T, ...T[]];

/**
 * ArrayToTuple converts an `Array<T>` to `[T, ...T]`
 *
 * eg: `[1, 2, 3]` from type `number[]` to `[number, number, number]`
 */
export type ArrayToTuple<Tuple extends NonEmptyArray<unknown>> = {
```